src/software/ai/navigator/navigator.cpp

```diff
@@ -317,19 +299,54 @@ double Navigator::getCloseToEnemyObstacleFactor(const Point &p)
         }
     }
 
-    // linear mapping of 0 to 1 onto 0 to 2 and return 1 when closest_dist is greater than
-    // 2
-    if (closest_dist > 2)
+    // clamp ratio between 0 and 1
```

Comment:
This isn't guaranteed to clamp to the range [0, 1] because the point (even assuming it's in the field) could be further than `ROBOT_MAX_SPEED`, which is only `2`. There's currently no relationship to "time" clear here so it doesn't make sense to create a ratio from `distance/speed`.

Reply:
ROBOT_MAX_SPEED was really acting as a stand in for the idea of "distance from neareset at which we don't need to slow down at all" so I made that a dynamic param